youtube_dl/extractor/zattoo.py

```diff
@@ -0,0 +1,259 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from uuid import uuid4
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    compat_str,
+    ExtractorError,
+    sanitized_Request,
+    urlencode_postdata,
+)
+
+
+class ZattooBaseIE(InfoExtractor):
+
+    _NETRC_MACHINE = 'zattoo'
+    _HOST_URL = 'https://zattoo.com'
+
+    _login_info = {}
+
+    def _login(self, uuid, session_id):
+        (username, password) = self._get_login_info()
+        if not username or not password:
+            raise ExtractorError(
+                'A valid %s account is needed to access this media.' % self._NETRC_MACHINE,
+                expected=True)
+        login_form = {
+            'login': username,
+            'password': password,
+            'remember': True,
+        }
+        request = sanitized_Request(
+            '%s/zapi/v2/account/login' % self._HOST_URL,
+            urlencode_postdata(login_form))
+        request.add_header(
+            'Referer', '%s/login' % self._HOST_URL)
+        request.add_header(
+            'Content-Type', 'application/x-www-form-urlencoded; charset=UTF-8')
+        request.add_header(
+            'Cookie', self._generate_cookie(uuid, session_id))
+        response = self._request_webpage(
+            request, None, 'Logging in')
+        cookie = response.headers.get('Set-Cookie')
+        pzuid = self._search_regex(r'pzuid\s*=\s*(.+?);', cookie, 'pzuid')
+        data = self._parse_json(response.read(), None)
+
+        return {
+            'ppid': data['session']['ppid'],
+            'powerhash': data['session']['power_guide_hash'],
+            'pzuid': pzuid,
+            'uuid': uuid,
+            'session_id': session_id
```

Comment:
Only `powerhash` is used. What's the point of mandatorily extracting and storing the rest?